### ISSUE TITLE
Enumerable#to_h with block

### ISSUE THREAD

**Issue by maisumakun** [marcandre/backports#126]
In Ruby 2.6, [Feature #15143](https://bugs.ruby-lang.org/issues/15143) was introduced, which accepts block for `Enumerable#to_h`.

```rb
# previously, we had to #to_h after #map.
(1..5).map{|x| [x, x ** 2]}.to_h

# can be done only #to_h
(1..5).to_h{|x| [x, x ** 2]}
```

In Ruby < 2.6, blocks for `Enumerable#to_h` seems to be simply ignored, so new functionality can be distinguished easily by `block_given?`.

**Comment by marcandre** [comment]
Indeed. Already working on that, not released yet... d33c9f5 and 18dd2e3

**Comment by marcandre** [comment]
Released in v3.12
